### PR TITLE
cli: deprecate use of react 16

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

In order to move #17414 and #7094 forward we'll be deprecating usage of React 16 in the 1.14 release. The goal is then to move to supporting both and only React 17 and React 18 in the 1.15 release in June.

If the `node_modules/react/packages.json` is at v16, the following warning will be printed when running `yarn start`:

image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by benjdlambert — APPROVED** [review]

**Review by jhaals — APPROVED** [review]
Nice!

File: packages/cli/src/commands/start/startFrontend.ts [hunk 1]
```diff
@@ -32,6 +32,30 @@ interface StartAppOptions {
...
+      paths: [paths.targetRoot],
+    });
+    const reactPkg = require(reactPkgPath);
+    if (reactPkg.version.startsWith('16.')) {
+      console.log(
+        chalk.yellow(
+          `
+⚠️                                                                           ⚠️
+⚠️ You are using React version 16, which is deprecated for use in Backstage. ⚠️
+⚠️ Please upgrade to at React 17 by updating your packages/app dependencies. ⚠️
```
Comment (freben) [inline packages/cli/src/commands/start/startFrontend.ts:48]:
```suggestion
⚠️ Please upgrade to at least React 17 by updating your packages/app dependencies. ⚠️
```

**Review by freben — COMMENTED** [review]
Right?

File: packages/cli/src/commands/start/startFrontend.ts (hunk 1 above)
Comment (freben) [reply packages/cli/src/commands/start/startFrontend.ts:48]: well at least (see what i did there?) remove the word "at" then.

Comment (Rugvip) [reply packages/cli/src/commands/start/startFrontend.ts]: Derp!

**Pull request merged**